src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -466,11 +466,11 @@ protected String generateAssertionsForPublicFields(Set<FieldDescription> fields,
   private String assertionContentForField(FieldDescription field, ClassDescription classDescription) {
     final String fieldName = field.getName();
     final String fieldNameCap = capitalize(field.getName());
-    if (classDescription.getGettersDescriptions().contains(new GetterDescription(fieldName, "get" + fieldNameCap,
-                                                                                 field.getTypeDescription(),
-                                                                                 Collections.<TypeName> emptyList()))) {
+    
+    if (classDescription.findGetterDescriptionForField(field) != null) {
```

Comment:
Additionally, I've realized that there is a _very_ real possibility of multiple. So either: 

1. return a `Set<>` of `Getter`s
2. Boolean on first found

I'm okay with either.